### ISSUE TITLE
Both 2LPT growth factor and source field are multiplied by 3/7

### ISSUE THREAD

**Issue by eelregit** [rainwoodman/fastpm-python#17]
I thought only one is needed?

**Comment by rainwoodman** [comment]
Only one is needed. I thought I checked that only 1 is applied though?

**Comment by eelregit** [comment]
Both the [D_2 ICs](https://github.com/rainwoodman/fastpm-python/blob/5bb126ccba28bff20a78b4ba4e1e51d2d87894ad/fastpm/background.py#L246) and [2LPT source](https://github.com/rainwoodman/fastpm-python/blob/5bb126ccba28bff20a78b4ba4e1e51d2d87894ad/fastpm/force/lpt.py#L63) have this factor, right?

How does the C version handle this?

**Comment by eelregit** [comment]
An unrelated question:
:)
In fastpm when pm_nc_factor is a list of numbers, e.g. {1, 2, 3} or {{0.0, 1}, {0.01, 2}}, what do they mean?

**Comment by rainwoodman** [comment]
Yes. Both have the factors, and C fastpm is also like that (which agreed
with the non-ode version, I think). So this is a serious, historical bug?

A list of numbers is in the format of a list of tuples of (scaling_factor,
new_pm_nc_factor). Switching pm_nc_factor with time.

**Comment by eelregit** [comment]
Not sure how serious that is though, given it's 2LPT and the limitation of PM accuracy.

Did you mean that the 2LPT agrees with PM to some tol so things should be fine?
Also I think there should be only one `3/7` but will need to confirm with other LPT experts on this.

**Comment by rainwoodman** [comment]
No I didn't mean 2LPT agrees with PM to some tol things should be fine. 

People have complained that the 2LPT from FastPM seem to have less small scale power than others, and this may be one of the contributing factors? 

I did check cFastPM with Martin's 2lpt IC in the early days (that they agree particle by particle), but things have drifted quite a bit since then. Also in the early days 2LPT was used in COLA mode, and I probably also checked that the results agreed with COLA.

Is there an established numerical result that we can compare with? Maybe we can compare the power spectrum of the divergence of DX2(z=0), under the definition where DX = DX1 + DX2.

**Comment by eelregit** [comment]
I guess a self-consistency test is to compare `PM(ZA(high z), lower z)` and `2LPT(lower z)`?
